tcms/xmlrpc/api/user.py

```diff
@@ -18,15 +18,14 @@
 
 
 def _get_user_dict(user):
-    u = XMLRPCSerializer(model=user)
-    u = u.serialize_model()
-    if 'password' in u:
-        del u['password']
-    return u
+    user_dict = XMLRPCSerializer(model=user).serialize_model()
+    if 'password' in user_dict:
+        del user_dict['password']
+    return user_dict
 
 
 @rpc_method(name='User.filter')
-def filter(query={}, **kwargs):
+def filter(query=None, **kwargs):  # pylint: disable=redefined-builtin
```

Comment:
Doesn't check if query is None and assign empty dict. Add a test for that use-case as well.